### PR TITLE
Fix some GpuBroadcastToRowExec by not dropping columns [databricks]

### PR DESCRIPTION
**Pull request by revans2**
This fixes another issue with AQE and GpuBroadcastToRowExec by not dropping columns when the rows are converted to unsafe rows. Any columns that are not needed will be dropped by the broadcast mode conversion anyways.

### CODE REVIEW

**Comment by revans2** [comment]
build

**Review by NVnavkumar — DISMISSED** [review]
LGTM

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastToRowExec.scala
Comment (jlowe) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastToRowExec.scala:1]: 2024 copyrights

File: integration_tests/src/main/python/aqe_test.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 # allow non gpu when time zone is non-UTC because of https://github.com/NVIDIA/spark-rapids/issues/9653'
 not_utc_aqe_allow=['ShuffleExchangeExec', 'HashAggregateExec'] if is_not_utc() else []
 
-_adaptive_conf = { "spark.sql.adaptive.enabled": "true" }
+_adaptive_conf = { "spark.sql.adaptive.enabled": "true"}
```
Comment (jlowe) [inline integration_tests/src/main/python/aqe_test.py:27]: Nit: Looks like an inadvertent change
```suggestion
_adaptive_conf = { "spark.sql.adaptive.enabled": "true" }
```

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
@jlowe @NVnavkumar please take another look

**Review by NVnavkumar — APPROVED** [review]

**Review by jlowe — APPROVED** [review]

**Pull request merged**